### PR TITLE
Add batch correction methods, Scanpy 1.6-specific things

### PR DESCRIPTION
**Pull request by pinin4fjords**
Predominantly the batch correction methods we're after:

 - Added [harmony](https://scanpy.readthedocs.io/en/latest/external/scanpy.external.pp.harmony_integrate.html) (see also https://github.com/slowkow/harmonypy/pull/8) https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/86/commits/85826e0078ed92b7884aeb3b77f68000230010b1
 - Added [BBKNN](https://scanpy.readthedocs.io/en/latest/external/scanpy.external.pp.bbknn.html) https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/86/commits/bbe1f524a34e22ef9881ee89aa51c3f777b1a949
 - Added [MNN](https://github.com/chriscainx/mnnpy) (bit fiddly due to interface) (see also https://github.com/bioconda/bioconda-recipes/pull/23893) https://github.com/ebi-gene-expression-group/scanpy-scripts/commit/0faa36948e028e224f0cf190f98342512d8ad790 
 - Added [ComBat](https://scanpy.readthedocs.io/en/latest/api/scanpy.pp.combat.html) https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/86/commits/924a74599ba74b3fcc4b51bede2118b800eeee1b
 - Removed bespoke behaviour for [hvg](https://scanpy.readthedocs.io/en/latest/api/scanpy.pp.highly_variable_genes.html), which seemed to have been written before very similar behaviour was present natively in Scanpy https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/86/commits/79ad11cf5c7fd2a0582f474d7690006756c93733. New seurat_v3 behaviour has implications for the batch handling and it seemed simpler to call function natively than it would have been to match it. 

I've tried to make the interfaces consistent where appropriate. In reference to previous patterns (if not entirely consistently with default behaviour in Scanpy itself), I've stored results in custom slots by default, e.g. layers, rather than at e.g. .X. Of course we could change the default behaviour to match Scanpy pretty easily.

Harmony became available in the latest Scanpy release (1.6.0).

### CODE REVIEW

File: scanpy_scripts/lib/_combat.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""
+scanpy combat
+"""
+
+import scanpy as sc
+
+# Wrapper for mnn allowing use of non-standard slot
+
+def combat(adata, key=None, key_added=None, layer=None, **kwargs):
```
Comment (pcm32) [inline scanpy_scripts/lib/_combat.py:9]: I'm guessing that this is using Combat_seq right? or is it applicable on differential expression results? My understanding is that the origina Combat was written for microarrays continuous real data and is not directly applicable to non-negative discrete count data.

File: setup.py [hunk 2]
```diff
@@ -41,12 +41,15 @@
         'matplotlib',
         'pandas',
         'h5py',
-        'scanpy>=1.5.1,<1.6.0',
```
Comment (pcm32) [inline setup.py:44]: do we plan to do a release with 1.5.x or will we just skip it?

**Comment by pcm32** [comment]
In principle it LGTM, but I guess that @nh3 is the most indicated person to review this.

File: scanpy_scripts/lib/_combat.py (hunk 1 above)
Comment (pinin4fjords) [reply scanpy_scripts/lib/_combat.py:9]: This is just a wrapper around Scanpy's own function: https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.combat.html, derived from https://github.com/brentp/combat.py.

People got very excited about the discrete thing with sequencing at the start, but my understanding is that subsequent testing indicted that in practice the issue was not such a big deal.

File: setup.py (hunk 2 above)
Comment (pinin4fjords) [reply setup.py:44]: Ahh- actually I should make this >=1.6.0- thanks.

Comment (pinin4fjords) [reply setup.py:44]: Some of the changes here are 1.6.0-specific, so will need this pin. I released a 1.5.1 compatible version yesterday (though without the batch stuff implemented here).

File: scanpy_scripts/cmds.py [hunk 3]
```diff
@@ -210,3 +214,31 @@
...
+
+COMBAT_CMD = make_subcmd(
+    'combat',
+    combat,
+    cmd_desc='ComBat function for batch effect correction',
+    arg_desc=_IO_DESC
+)
+
+HARMONY_INTEGRATE_CMD = make_subcmd(
+    'harmony_integrate',
```
Comment (nh3) [inline scanpy_scripts/cmds.py:226]: I feel "harmony_integrate" is a bit long. I might have missed something but is "harmony" ok?

File: scanpy_scripts/cmds.py [hunk 4]
```diff
@@ -210,3 +214,31 @@
...
+
+BBKNN_CMD = make_subcmd(
+    'bbknn',
+    bbknn,
+    cmd_desc='Batch balanced kNN [Polanski19].',
+    arg_desc=_IO_DESC,
+)
+
+MNN_CORRECT_CMD = make_subcmd(
+    'mnn_correct',
```
Comment (nh3) [inline scanpy_scripts/cmds.py:240]: Similarly, can "mnn_correct" be shortened to "mnn"? Just for lazy command line users...

**Review by nh3 — APPROVED** [review]
LGTM too. Great job again!

File: scanpy_scripts/cmds.py (hunk 3 above)
Comment (pinin4fjords) [reply scanpy_scripts/cmds.py:226]: There's also 'harmony-timeseries' in Scanpy, so I thought I'd use the full function name to be clear. But I guess being a subfunction of integrate should be clear enough

**Comment by pinin4fjords** [comment]
Thanks @nh3 @pcm32 !

**Pull request merged**